src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        int arraySize = 6; // Можна змінити на інший розмір
+        Figure[] figures = new Figure[arraySize];
+
+        // Заповнення першої половини масиву випадковими фігурами
+        for (int i = 0; i < arraySize / 2; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        // Заповнення другої половини масиву стандартними фігурами
+        for (int i = arraySize / 2; i < arraySize; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
```

Comment:
Instead of dividing 'arraySize' by 2 in both loop conditions, consider storing the result in a variable for better readability and to avoid repeating the calculation.